picard/ui/scripteditor.py

```diff
@@ -0,0 +1,554 @@
+        """
+        selected = self.ui.preset_naming_scripts.currentIndex()
+        selected_script = PRESET_SCRIPTS[selected]['script']
+        if selected_script is None:
+            config = get_config()
+            self.set_script(config.setting['file_naming_format'])
+        else:
+            self.set_script(selected_script)
+        self.update_examples()
+
+    @staticmethod
+    def synchronize_selected_example_lines(current_row, source, target):
+        """Matches selected item in target to source"""
+        if source.currentRow() != current_row:
+            current_row = source.currentRow()
+            target.blockSignals(True)
+            target.setCurrentRow(current_row)
+            target.blockSignals(False)
+
+    def match_after_to_before(self):
+        """Sets the selected item in the 'after' list to the corresponding item in the 'before' list.
+        """
+        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_before, self.ui.example_filename_after)
+
+    def match_before_to_after(self):
+        """Sets the selected item in the 'before' list to the corresponding item in the 'after' list.
+        """
+        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_after, self.ui.example_filename_before)
+
+    def save_script(self):
+        """Emits a `save` signal to trigger appropriate save action in the parent object.
+        """
+        self.signal_save.emit()
+
+    def get_script(self):
+        """Provides the text of the file naming script currently loaded into the editor.
+
+        Returns:
+            str: File naming script
+        """
+        return str(self.ui.file_naming_format.toPlainText()).strip()
+
+    def set_script(self, script_text):
+        """Sets the text of the file naming script into the editor.  Sets default text if `script_text` is empty or missing.
+
+        Args:
+            script_text (str): File naming script text to set in the editor.
+        """
+        self.ui.file_naming_format.setPlainText(str(script_text).strip())
+
+    def update_example_files(self):
+        """Update the before and after file naming examples list.
+        """
+        self.examples.update_sample_example_files()
+        self.display_examples()
+
+    def update_examples(self):
+        """Update the before and after file naming examples using the current file naming script in the editor.
+        """
+        override = {'file_naming_format': self.get_script()}
+        self.examples.update_examples(override)
+        self.display_examples()
+
+    @staticmethod
+    def update_example_listboxes(before_listbox, after_listbox, examples):
+        before_listbox.clear()
+        after_listbox.clear()
+        for before, after in sorted(examples, key=lambda x: x[1]):
+            before_listbox.addItem(before)
+            after_listbox.addItem(after)
+
+    def display_examples(self, send_signal=True):
```

Comment:
I don't think we ever have `send_signal == False` in current code. Perhaps this option can be removed.

Reply:
Good catch!  That was a leftover from a previous version of the code.  You're right, it's no longer used, so I will remove it.  Thanks.